### PR TITLE
fix: bunch of bugs on macOS platform

### PR DESCRIPTION
**Pull request by harshavardhana**
fixes for all warnings reported by clang

### CODE REVIEW

**Comment by balamurugana** [comment]
Can you enable github workflow on `macos`?

File: src/http.cc [hunk 1]
```diff
@@ -207,6 +207,9 @@ minio::http::Response minio::http::Request::execute() {
   std::istream body_stream(&charbuf);
 
   switch (method) {
+    case Method::kDelete:
+    case Method::kGet:
+      break;
```
Comment (balamurugana) [inline src/http.cc:212]: Is it rneeded for `macos`?

Comment (harshavardhana) [reply src/http.cc:212]: its a `clang` complaint @balamurugana to make sure to all `enums` are present in the `switch`

**Comment by harshavardhana** [comment]
> Can you enable github workflow on `macos`?

Yes, I will try to set it up once this PR is merged.

File: check-style.sh [hunk 2]
```diff
@@ -10,8 +11,8 @@ function clang_format() {
 }
 
 ec=0
-mapfile -t files < <(find src include examples tests -iname "*.cc" -o -iname "*.h")
-for file in "${files[@]}"; do
+source_files=$(find src include examples tests -type f \( -name '*.h' -o -name '*.cc' \))
```
Comment (balamurugana) [inline check-style.sh:14]: This will break for file names containing spaces.

Comment (harshavardhana) [reply check-style.sh:14]: Why would you create filenames with spaces? @balamurugana

Comment (harshavardhana) [reply check-style.sh:14]: Also main reason for this change is that bash shipped by default on macOS  doesn't support for `mapfile`

Comment (balamurugana) [reply check-style.sh:14]: Output of `find` command will output file names on which non-project files will be listed and those file names might have spaces. If `mapfile` is not available, we would need to create a temporary file for `find` output to work reliably.

```bash
tmpfile="tmpfile.$RANDOM"
find src include examples tests -iname "*.cc" -o -iname "*.h" > "$tmpfile"
ec=0
while read -r file; do
    if ! clang_format "$file"; then
        ec=255
    fi
done < "$tmpfile"
rm -f "$tmpfile"
```

Comment (harshavardhana) [reply check-style.sh]: done updated using the `filename` approach

**Comment by harshavardhana** [comment]
Looks like all good everything works here @balamurugana - next is to get Windows support in.

**Review by balamurugana — APPROVED** [review]

**Pull request merged**